client/lib/plans/index.js

```diff
@@ -57,3 +57,15 @@ export function isInGracePeriod( plan ) {
 export function shouldFetchSitePlans( sitePlans, selectedSite ) {
 	return ! sitePlans.hasLoadedFromServer && ! sitePlans.isFetching && selectedSite;
 };
+
+export function filterPlansBySiteAndProps( plan, site, hideFreePlan ) {
+	if ( site && site.jetpack ) {
+		return 'jetpack_business' === plan.product_slug || 'jetpack_premium' === plan.product_slug;
```

Comment:
Whenever possible, I think we should keep product slug values in `lib/products-values`. We can add a function, `isJetpackPlan`, and use an existing function, `isFreePlan`, to obviate strings like `'jetpack_premium'` in this function altogether:

```js
function filterPlansBySiteAndProps( plan, site, hideFreePlan ) {
  if ( site && site.jetpack ) {
    return isJetpackPlan( plan );
  }

  if ( hideFreePlan && isFreePlan( plan ) ) {
    return false;
  }

  return ! isJetpackPlan( plan );
}
```